src/commands/login/server/index.ts

```diff
@@ -0,0 +1,109 @@
+import express, { ErrorRequestHandler } from "express";
+import { Server } from "http";
+import { BAD_REQUEST, OK } from "http-status-codes";
+import { Socket } from "net";
+
+import { CLIError } from "../../../errors";
+import logger from "../../../util/logger";
+
+export class LoginAuthServer {
+    redirectPath: string;
+    stopped = false;
+    server: Server | undefined;
+    connections: Socket[] = [];
+    accessToken: string | undefined;
+
+    constructor(redirectPath: string) {
+        this.redirectPath = redirectPath;
+    }
+
+    start(port: number): Promise<Server> {
+        if (this.server?.listening) {
+            return Promise.resolve(this.server);
+        }
+
+        const app = express();
+
+        // CORS
+        app.use((_req, res, next) => {
+            res.header("Access-Control-Allow-Origin", "*");
+            res.header("Access-Control-Allow-Headers", "X-Requested-With, Origin");
+            res.header("Access-Control-Allow-Methods", "GET");
+            next();
+        });
+
+        app.get(this.redirectPath, async (req, res, next) => {
+            this.accessToken = req.query.access_token;
+
+            if (!this.accessToken) {
+                next(new Error("No access token!"));
+            } else {
+                res.status(OK).json({ accessToken: this.accessToken });
+
+                await this.stop();
+            }
+        });
+
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const errorHandler: ErrorRequestHandler = async (err, req, res, _next) => {
+            res.status(BAD_REQUEST).json({ error: err.message });
+
+            if (req.path === this.redirectPath) {
+                await this.stop();
+            }
+        };
```

Comment:
```suggestion
        app.get(this.redirectPath, async (req, res, next) => {
            this.accessToken = req.query.access_token;

            if (!this.accessToken) {
                res.status(BAD_REQUEST).json({ error:"No access token" });
            } else {
                res.status(OK).json({ accessToken: this.accessToken });
            }
            await this.stop();
        });
        
        
        export const handleError: ErrorRequestHandler = (err, req, res, next) => {
            if (!res.headersSent) {
                res.status(err?.statusCode ?? INTERNAL_SERVER_ERROR).json({
                    detail: err?.message ?? "Unexpected Error",
                    title: err?.title ?? "Unexpected Error"
                });
            }
            next(err);
        };
```

It's better to keep the error handler more generic by moving stop logic from error handler to controller of redirectPath. We can also implement the function without disabling eslint.